src/test/scala/uk/gov/nationalarchives/tdr/api/routes/ConsignmentRouteSpec.scala

```diff
@@ -162,15 +166,46 @@ class ConsignmentRouteSpec extends AnyFlatSpec with Matchers with TestRequest wi
     response.errors.head.message should equal(expectedResponse.errors.head.message)
   }
 
-  private def checkConsignmentExists(consignmentId: UUID): Unit = {
+  "updateExportLocation" should "update the export location correctly" in {
+    createConsignment(new FixedUUIDSource().uuid, userId)
+    val prefix = "json/updateexportlocation_"
+    val expectedResponse = getDataFromFile[GraphqlMutationExportLocation](prefix)("data_all")
+    val token = validBackendChecksToken("export")
+    val response: GraphqlMutationExportLocation = runTestRequest[GraphqlMutationExportLocation](prefix)("mutation_all", token)
+    response.data should equal(expectedResponse.data)
+    getConsignmentField(UUID.fromString("6e3b76c4-1745-4467-8ac5-b4dd736e1b3e"), "ExportLocation") should equal("6e3b76c4-1745-4467-8ac5-b4dd736e1b3e.tar.gz")
+  }
+
+  "updateTransferInitiated" should "update the transfer initiated date correctly" in {
+    createConsignment(new FixedUUIDSource().uuid, userId)
+    val prefix = "json/updatetransferinitiated_"
+    val expectedResponse = getDataFromFile[GraphqlMutationTransferInitiated](prefix)("data_all")
+    val response: GraphqlMutationTransferInitiated = runTestRequest[GraphqlMutationTransferInitiated](prefix)("mutation_all", validUserToken())
+    response.data should equal(expectedResponse.data)
+    val field = getConsignmentField(UUID.fromString("6e3b76c4-1745-4467-8ac5-b4dd736e1b3e"), _)
+    Option(field("TransferInitiatedDatetime")).isDefined should equal(true)
+    field("TransferInitiatedBy") should equal(userId.toString)
+  }
+
+  private def getConsignment(consignmentId: UUID) = {
     val sql = s"select * from Consignment where ConsignmentId = ?"
     val ps: PreparedStatement = DbConnection.db.source.createConnection().prepareStatement(sql)
     ps.setString(1, consignmentId.toString)
-    val rs: ResultSet = ps.executeQuery()
+    ps.executeQuery()
+  }
+
+  private def checkConsignmentExists(consignmentId: UUID): Unit = {
+    val rs = getConsignment(consignmentId)
     rs.next()
     rs.getString("ConsignmentId") should equal(consignmentId.toString)
   }
 
+  private def getConsignmentField(consignmentId: UUID, field: String): String = {
+    val rs = getConsignment(consignmentId)
+    rs.next()
```

Comment:
Is it possible to move the `rs.next()` into the method that executes the query? It's a bit confusing when it's out of context here.

Also, I'd name the variable `result` or something - `rs` is also quite opaque when it's away from the context of a `ResultSet`.

Reply:
Yep, fair point, I've changed it.